client/cmd/flags.go

```diff
@@ -377,3 +657,88 @@ func validateCommissionRate(ctx context.Context, cfg *createValidatorConfig) err
 
 	return nil
 }
+
+func validateNewCommissionRate(ctx context.Context, cfg *updateCommissionConfig, pubKey []byte) error {
+	validator, err := getValidatorByEVMAddr(ctx, cfg.StakingAPI, pubKey)
+	if err != nil {
+		return err
+	}
+
+	if validator.OperatorAddress == "" {
+		return errors.New("validator not found")
+	}
+
+	prevCommission := validator.Commission
+	newCommission := math.LegacyNewDecWithPrec(int64(cfg.CommissionRate), 4)
+
+	if time.Since(prevCommission.UpdateTime).Hours() < 24 {
+		return stypes.ErrCommissionUpdateTime
+	}
+
+	if newCommission.GT(prevCommission.MaxRate) {
+		return stypes.ErrCommissionGTMaxRate
+	}
+
+	if newCommission.Sub(prevCommission.Rate).GT(prevCommission.MaxChangeRate) {
+		return stypes.ErrCommissionGTMaxChangeRate
+	}
+
+	return nil
+}
+
+type Validator struct {
+	OperatorAddress string            `json:"operator_address"`
+	Commission      stypes.Commission `json:"commission"`
+}
```

Comment:
Instead of using `stypes.Validator`, redefined a simplified Validator struct and used here. Since the `unbonding_height` is returned in string, but it is int64 in `stypes.Validator`.